### DISCUSSION TITLE
How can I use "useArrayField" if I have several arrays in one form?

### DISCUSSION THREAD

**Discussion by REDLIGHTFILMS** [react-hook-form/react-hook-form#1959] [Q&A]
My problem is I want to have multiple array fields in my form and solve this with "useArrayField" how can I do that, because I can't give multiple names.

Here is my data structure:

```
personalData:{
  numberOfEmployees: [
    {year: 2012, quantity: 2000}
  ],
  bosses: [
    {name: "Oliver", taskArea: "CEO", active: true}
  ],
  singleContacts: [
    {name: "James", taskArea: "CTO"}
  ]
}
```

I have something like that right there:

```
const { fields, append, remove } = useFieldArray({
      control,
      name: "contactData.numberOfEmployees"
});
```

But how can I represent the other arrays So then you have to duplicate useArrayField or how do you do that?Because just typing in contactdata does not work. 

How can I solve this?

**Comment by bluebill1049** [comment]
multiple `useFieldArray` instance

**Reply by REDLIGHTFILMS** [reply]
How do I do that?

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-x7btr
